### PR TITLE
feat(commits): added linter to enforce conventional commit messages

### PR DESCRIPTION
**Pull request by kleyow**

### CODE REVIEW

File: commitlint.config.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      ['sentence-case', 'start-case', 'pascal-case', 'upper-case']
+    ],
+    'subject-empty': [2, 'never'],
+    'subject-full-stop': [2, 'never', '.'],
+    'type-case': [2, 'always', 'lower-case'],
+    'type-empty': [2, 'never'],
+    'type-enum': [
+      2,
+      'always',
+      [
```
Comment (eoln) [inline commitlint.config.js:22]: wouldn't it better to import the commit types from the list specified already in https://github.com/mojaloop/thirdparty-api-adapter/blob/master/.versionrc.js
then we will have only one source of truth for commit types

Comment (kleyow) [reply commitlint.config.js]: done

File: commitlint.config.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+const versionrc = require('./.versionrc.js')
+
+const types = versionrc.types.map(function(typeDict) {
```
Comment (eoln) [inline commitlint.config.js:3]:
```suggestion
const types = versionrc.types.map((typeDict) => typeDict['type'])
```

File: .eslintignore [hunk 3]
```diff
@@ -1,3 +1,4 @@
 node_modules/**/*.js
 coverage/*
-.circleci/*
\ No newline at end of file
+.circleci/*
+commitlint.config.js
```
Comment (eoln) [inline .eslintignore:4]: I think we need this file linted

File: commitlint.config.js (hunk 2 above)
Comment (kleyow) [reply commitlint.config.js]: done

File: .eslintignore (hunk 3 above)
Comment (kleyow) [reply .eslintignore]: done

**Pull request merged**